src/graph/backend/elyzor/include/dnnl_graph_compiler.h

```diff
@@ -15,9 +16,41 @@
 extern "C" {
 #endif
 
+// graph compiler's API version following semver
+#define DNNL_GC_API_V_MAJOR 0
+#define DNNL_GC_API_V_MINOR 1
+#define DNNL_GC_API_V_PATCH 0
+#ifdef DNNL_VERSION_HASH
+#define DNNL_GC_API_V_HASH DNNL_VERSION_HASH
+#else
+#define DNNL_GC_API_V_HASH "N/A"
+#endif
+
 struct dnnl_graph_compiler;
 struct dnnl_graph_compiler_executable;
 
+struct dnnl_graph_compiler_version {
+    struct version {
+        uint8_t major;
+        uint8_t minor;
+        uint8_t patch;
+        const char *hash;
+    };
+    // version of the gc API that was used to compile gc,
+    // we only bump it if the API in this file changes
+    version api_version;
+    // version of the graph compiler itself, we only bump it if
+    // the core graph compiler releases a new version (with new
+    // compiling features, etc) that does not necessarily change
+    // the API in this file
+    version gc_version;
+    // Why having two versions?
+    // The versions are bumped independently, so we want to know both versions of the '.so':
+    //     1. API version - to check whether it's compatible at all with the current oneDNN version.
+    //     2. GC version - to know which features/compilation patterns are supported in order to
+    //        dispatch backend's logic based on this information (register certain patterns or not)
```

Comment:
_Originally posted by @AndreyPavlenko in https://github.com/intel-ai/oneDNN/pull/5#discussion_r1592661715_

What if instead of the GC version use the feature flags? E.g.:
```c++
bool has_feature_a;
bool has_feature_b;
...
```
Writing code that makes assumptions on a thirdparty library version is error prone and requires additional efforts for the code support. A simple example:
```c++
if (gc.version.major >= 1 && gc.version.minor >= 1) {
  // Using feature F, that is supported by GC since the version 1.1.0
}
```
Assume that the feature F has been removed in the version 2.0.0. Now the new version of the GC library is not compatible with the previous versions of oneDNN.

Reply:
_Originally posted by @dchigarev in https://github.com/intel-ai/oneDNN/pull/5#discussion_r1592790839_

> What if instead of the GC version use the feature flags? E.g.:

The feature flags could actually be useful. However, I wouldn't say we should use them instead of GC version, but rather together as writing each optimization pass to the feature flags seems too much. If needed, the feature flags can be added as an additional meta info to the `dnnl_graph_compiler_version` structure [(like this)](https://github.com/intel-ai/oneDNN/pull/5#discussion_r1592380197)

Anyway, currently we don't have a list of features that we would like to put here (@kurapov-peter correct me if I'm wrong), so I would propose to leave everything as it is and add feature flags later if we'll find a need for them.